Editor/Window/GameLiftRequestAdapter.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Amazon.GameLift;
+using Amazon.GameLift.Model;
+using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Editor.Window
+{
+    public interface IAmazonGameLiftClientFactory
+    {
+        IAmazonGameLiftClientWrapper Get(string profile);
+    }
+
+    public class AmazonGameLiftClientFactory : IAmazonGameLiftClientFactory
+    {
+        private readonly CoreApi _coreApi;
+
+        public AmazonGameLiftClientFactory(CoreApi coreApi)
+        {
+            _coreApi = coreApi;
+        }
+
+        public IAmazonGameLiftClientWrapper Get(string profile)
+        {
+            var credentials = _coreApi.RetrieveAwsCredentials(profile);
+            var client = new AmazonGameLiftClient(credentials.AccessKey, credentials.SecretKey);
+            return new AmazonGameLiftWrapper(client);
+        }
+    }
+
+    public class GameLiftRequestAdapter
+    {
+        private GameLiftPlugin _gameLiftConfig;
```

Comment:
This dependency doesn't make sense to me. The GameLiftPlugin is the top-level editor window and should only handle the top level editor functionality, it has nothing to do with the gamelift wrapper and core api code. It seems like there's a better solution for getting the wrapper and core api other than adding them to GameLiftPlugin and delegating in this class

Reply:
I have changed how this works, rather than handing the editor window to every class, I instead have a StateManager that keeps a track of this and is shared between the classes instead.